### PR TITLE
Better handle request expiration and additional clean up

### PR DESCRIPTION
**Pull request by akrambek**

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCacheEntry.java [hunk 1]
```diff
@@ -57,7 +57,6 @@
 
     private BufferPool requestPool;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCacheEntry.java]: Should be `final` ?

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java [hunk 2]
```diff
@@ -473,29 +479,29 @@ private void retryCacheableRequest()
     private boolean storeRequest(
         final ListFW<HttpHeaderFW> headers)
     {
-        this.requestSlot = factory.requestBufferPool.acquire(acceptInitialId);
-        if (requestSlot == NO_SLOT)
+        this.requestSlot.value = factory.requestBufferPool.acquire(acceptInitialId);
+        if (requestSlot.value == NO_SLOT)
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java]: Recommend you add an `assert requestSlot.value == NO_SLOT;` before the `acquire`, then assign to local variable `newRequestSlot` first, then only assign to `requestSlot.value` if `newRequestSlot != NO_SLOT`.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java [hunk 3]
```diff
@@ -473,29 +479,29 @@ private void retryCacheableRequest()
...
 
     private void purge()
     {
-        if (requestSlot != NO_SLOT)
+        if (requestSlot.value != NO_SLOT)
         {
-            factory.requestBufferPool.release(requestSlot);
-            this.requestSlot = NO_SLOT;
+            factory.requestBufferPool.release(requestSlot.value);
+            this.requestSlot.value = NO_SLOT;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java]: Remove `this.`

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCacheEntry.java (hunk 1 above)
Comment (akrambek) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCacheEntry.java]: Done.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java (hunk 2 above)
Comment (akrambek) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java]: Done.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java (hunk 3 above)
Comment (akrambek) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java]: Done.

**Pull request merged**